pkg/controllers/provisioning/provisioner.go

```diff
@@ -337,6 +337,9 @@ func (p *Provisioner) Schedule(ctx context.Context) (scheduler.Results, error) {
 	if err != nil {
 		if errors.Is(err, ErrNodePoolsNotFound) {
 			log.FromContext(ctx).Info("no nodepools found")
+			for _, pod := range pods {
+				p.cluster.ClearPodSchedulingMappings(client.ObjectKeyFromObject(pod))
```

Comment:
nit: Consider updating the `ClearPodSchedulingMappings` so that it takes a variadic

Reply:
Can we consider adding a metric test here or minimally posting the results from pulling the metrics from these pods after the update?